feature: adding support for new relic and lighstep

xoscar opened the pull request:
This PR adds support for both new relic and light step to show up as data store options under the settings page.

## Changes

- Updates the UI to show a different input for the list
- Adds new relic and lightstep

## Fixes

- #1793 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/4c289d7351f846029d2521958029a15b

jorgeepc commented on server/model/data_stores.go:
```diff
@@ -77,6 +79,8 @@ func DataStoreFromConfig(dsc config.TracingBackendDataStoreConfig) DataStore {
 		ds.Type = openapi.OPEN_SEARCH
 	case signalfx:
 		ds.Type = openapi.SIGNAL_FX
+	case newrelic:
+	case lighstep:
```
If you have `newrelic` or `lighstep` in your config file, is this going to change the type to `otlp`? Do we want that behavior?

xoscar replied:
Nice catch, fixed!

jorgeepc reviewed (commented):
Looks great. I added one question and a minor suggestion: I would put `OpenSearch` and `Elastic` together in the list.

xoscar commented:
> Looks great. I added one question and a minor suggestion: I would put `OpenSearch` and `Elastic` together in the list.

I added them based on the list from the designs, I'll put them together 😄
